packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.tsx

```diff
@@ -79,7 +71,28 @@ function formatValue(key: string, value: any) {
   } else if (Array.isArray(parsed) && shouldDisplayAsStringList(key)) {
     content = stringListMarkup(parsed);
   } else {
-    content = _jsonMarkup(parsed);
+    const shouldJsonTreeExpand = Object.keys(parsed).length <= 10;
```

Comment:
I think this is good as a first approximation, but the right solution would be estimating the total height to be >10, not just the number of first-level children.

Reply:
Oh, I guess we could work on some algorithm, It could be something like 1 property = 8 px, and then we could just go on calculating the height of the first X elements (recursively) and then decide to expand or collapse the tree

This task could be included in the upcoming LFX term